### PR TITLE
Refactor/vuex classes refactor

### PR DESCRIPTION
**Pull request by VoroniakPavlo**

### CODE REVIEW

File: src/store/modules/contact-center/agents/agents.js [hunk 1]
```diff
@@ -8,162 +7,58 @@ import {
     addAgent, deleteAgent,
     getAgent, getAgentsList, updateAgent,
 } from "../../../../api/contact-center/agents/agents";
+import {DefaultModule} from "../../defaults/DefaultModule";
+import proxy from "../../../../utils/editProxy";
```
Comment (VoroniakPavlo) [inline src/store/modules/contact-center/agents/agents.js]: rm unused import

File: src/store/modules/routing/gateways/gateways.js [hunk 2]
```diff
@@ -88,38 +86,6 @@ const actions = {
...
-    PREV_PAGE: (context) => {
-        if(state.page) {
-            context.commit('DECREMENT_PAGE');
-            context.dispatch('LOAD_DATA_LIST');
-        }
-    },
-
     TOGGLE_ITEM_PROPERTY: async (context, index) => {
         await context.commit('TOGGLE_ITEM_PROPERTY', index);
         let changes = {enable: state.dataList[index].enable};
```
Comment (VoroniakPavlo) [inline src/store/modules/routing/gateways/gateways.js]: patch

File: src/store/modules/routing/dialplan/dialplan.js [hunk 3]
```diff
@@ -109,70 +75,10 @@ const actions = {
...
-    INCREMENT_PAGE: (state) => {
-        state.page++;
-    },
-
-    DECREMENT_PAGE: (state) => {
-        state.page--;
-    },
 
     TOGGLE_ITEM_PROPERTY: (state, index) => {
         state.dataList[index].disabled = !state.dataList[index].disabled;
```
Comment (VoroniakPavlo) [inline src/store/modules/routing/dialplan/dialplan.js]: patch

File: src/store/modules/directory/users/users.js [hunk 4]
```diff
@@ -59,38 +57,6 @@ const actions = {
...
-    PREV_PAGE: (context) => {
-        if(state.page) {
-            context.commit('DECREMENT_PAGE');
-            context.dispatch('LOAD_DATA_LIST');
-        }
-    },
-
     PATCH_ITEM_PEROPERTY: async (context, {value, index}) => {
         await context.commit('PATCH_ITEM_PEROPERTY', {value, index});
         let changes = {status: value};
```
Comment (VoroniakPavlo) [inline src/store/modules/directory/users/users.js]: patch

File: src/store/modules/contact-center/resources/resources.js [hunk 5]
```diff
@@ -119,72 +66,21 @@ const actions = {
...
-        state.page--;
-    },
-
-    SET_ITEM_PROPERTY: (state, {prop, value}) => {
-        state.itemInstance[prop] = value;
-    },
 
     TOGGLE_ITEM_PROPERTY: (state, {index, prop}) => {
         state.dataList[index][prop] = !state.dataList[index][prop];
     },
```
Comment (VoroniakPavlo) [inline src/store/modules/contact-center/resources/resources.js]: patch

File: src/components/integrations/storage/the-storage.vue [hunk 6]
```diff
@@ -71,7 +71,7 @@
                 <template slot="enabled" slot-scope="props">
                     <switcher
                             :value="!dataList[props.rowIndex].disabled"
-                            @input="toggleItemProperty(props.rowIndex)"
+                            @input="patchProperty({index: props.rowIndex, prop: 'disabled', value: $event})"
```
Comment (dlohvinov) [inline src/components/integrations/storage/the-storage.vue]: add default object with disabled: false to list getter

File: src/components/routing/dialplan/the-dialplan.vue [hunk 7]
```diff
@@ -138,6 +138,7 @@
 
         watch: {
             dataList() {
+
```
Comment (dlohvinov) [inline src/components/routing/dialplan/the-dialplan.vue]: за шо

File: src/store/modules/defaults/DefaultModule.js [hunk 8]
```diff
@@ -55,6 +55,19 @@ export class DefaultModule {
                 context.commit('SET_ITEM_PROPERTY', payload);
             },
 
+            PATCH_ITEM_PROPERTY: async (context, {index, prop, value}) => {
+                console.log({value, prop, index});
```
Comment (dlohvinov) [inline src/store/modules/defaults/DefaultModule.js]: rm console.log

File: src/store/modules/defaults/DefaultModule.js [hunk 9]
```diff
@@ -117,6 +130,12 @@ export class DefaultModule {
                 state.itemInstance = item;
             },
 
+            PATCH_ITEM_PROPERTY: (state, {index, prop, value}) => {
+                console.log(state.dataList[index][prop])
+                state.dataList[index][prop] = value;
+                console.log(state.dataList[index][prop])
```
Comment (dlohvinov) [inline src/store/modules/defaults/DefaultModule.js]: rm console.log

**Pull request merged**